packages/backend/src/api/controllers/ActivityController.ts

```diff
@@ -22,46 +23,62 @@ export class ActivityController {
   ) {}
 
   async getTransactionActivity(): Promise<ActivityApiResponse> {
-    const [layer2Counts, ethereumCounts] = await Promise.all([
-      this.getLayer2Counts(),
-      this.ethereumCounter.getDailyTransactionCounts(),
+    const [layer2s, ethereumCounts] = await Promise.all([
+      this.getLayer2s(),
+      this.ethereumCounter.getFullySyncedDailyCounts(),
     ])
+    const tip = this.getTip(layer2s, ethereumCounts)
+    const fullySyncedLayer2s = layer2s.map(({ counts, projectId }) => ({
+      projectId,
+      counts: this.limitCounts(counts, tip),
+    }))
+    const fullySyncedEthereumCounts = this.limitCounts(ethereumCounts, tip)
 
     return {
-      combined: this.toCombinedActivity(layer2Counts),
-      projects: this.toProjectsActivity(layer2Counts),
-      ethereum: this.countsToChart(ethereumCounts),
+      combined: this.toCombinedActivity(fullySyncedLayer2s),
+      projects: this.toProjectsActivity(fullySyncedLayer2s),
+      ethereum: this.countsToChart(fullySyncedEthereumCounts),
     }
```

Comment:
All of this, including the private functions is pure. It would be easier to write and test if it was extracted outside of the class.

Also the tests for this are really lacking, once its extracted it would be nice to see that the tip is calculated correctly for example.